crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -194,4 +194,151 @@ mod advancing_time_tests {
         let result = devnet.post_json("/increase_time".into(), increase_time_body).await.unwrap();
         assert_eq!(result.status(), 422);
     }
+
+    #[tokio::test]
+    async fn wrong_start_time() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--start-time", "wrong"]).await;
+        assert!(devnet.is_err());
+    }
+
+    #[tokio::test]
+    async fn start_time_in_past() {
+        let now = get_unix_timestamp_as_seconds();
+        let past_time = 1;
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--start-time",
+            past_time.to_string().as_str(),
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        // create block and check if block timestamp is greater/equal 1 and less than now
+        devnet.post_json("/create_block".into(), Body::from(json!({}).to_string())).await.unwrap();
+        let empty_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+        assert!(empty_block["timestamp"].as_u64() >= Some(past_time));
+        assert!(empty_block["timestamp"].as_u64() < Some(now));
+    }
+
+    #[tokio::test]
+    async fn start_time_in_future() {
+        let now = get_unix_timestamp_as_seconds();
+        let future_time = now + 100;
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--start-time",
+            future_time.to_string().as_str(),
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        // create block and check if block timestamp is greater than now
+        devnet.post_json("/create_block".into(), Body::from(json!({}).to_string())).await.unwrap();
+        let empty_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+        assert!(empty_block["timestamp"].as_u64() > Some(now));
```

Comment:
I don't get this comment. I set cli start-time in the future and mine empty block to check if it's greater than now.

If I comment start time code this test will fail so for me it looks like the test is working as expected.

Reply:
I was trying to say that if somehow devnet is not working properly by not properly increasing the time (e.g. by 50 instead of 100), the check would still pass. So instead of doing
```rust
assert!(empty_block["timestamp"].as_u64() > Some(now));
```
why not do
```rust
 assert!(empty_block["timestamp"].as_u64() >= Some(future_time));
```

Of course, with keeping the newly added buffered upper limit.

Reply:
Fixed in 3ad7a39232ba2bb7ed2dce39ae61dade4685b31e